File: iosApp/iosApp/feature/motivation/MotivationDataMapper.swift
//
//  MotivationDataMapper.swift
//  iosApp
//
//  Created by Марк Шавловский on 22.06.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared

extension MotivationData{
    func getMotivationUi() -> MotivationUi? {
            switch self {
            case let minOrderCost as MotivationDataMinOrderCost:
                return MotivationUi.MinOrderCost(minOrderCost.cost)
            case let forLowerDelivery as MotivationDataForLowerDelivery:
                return MotivationUi.ForLowerDelivery(
                    forLowerDelivery.increaseAmountBy,
                    forLowerDelivery.progress,
                    forLowerDelivery.isFree
                )
            case let lowerDeliveryAchieved as MotivationDataLowerDeliveryAchieved:
                return MotivationUi.LowerDeliveryAchieved(lowerDeliveryAchieved.isFree)
            default:
                return nil
            }
    }
}

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/ui/motivation/MotivationDataMapper.kt
package com.bunbeauty.shared.ui.motivation

import com.bunbeauty.shared.presentation.motivation.MotivationData

fun MotivationData.toMotivationUi(): MotivationUi =
    when (this) {
        is MotivationData.MinOrderCost -> {
            MotivationUi.MinOrderCost(cost = cost)
        }

        is MotivationData.ForLowerDelivery -> {
            MotivationUi.ForLowerDelivery(
                increaseAmountBy = increaseAmountBy,
                progress = progress,
                isFree = isFree,
            )
        }

        is MotivationData.LowerDeliveryAchieved -> {
            MotivationUi.LowerDeliveryAchieved(isFree = isFree)
        }
    }
